views/js/controller/creator/helpers/validators.js

```diff
@@ -0,0 +1,200 @@
+                    if (typeof callback === 'function') {
+                        const counts = _.countBy(identifiers, 'identifier');
+                        //the identifier list contains itself after change on input
+                        //on keyup $idInUI.length === 0
+                        //on change and blur $idInUI.length === 1 and text equal value
+                        callback(
+                            typeof counts[key] === 'undefined' ||
+                            $idInUI.length === 1 && $idInUI.text() === value && counts[key] === 1
+                        );
+                    }
+                } else {
+                    throw new Error('missing required option "identifier"');
+                }
+            }
+        };
+    }
+
+    /**
+     * Gives you a validator that check if a QTI id is available
+     * @param {Object} modelOverseer - let's you get the data model
+     */
+    function registerValidators(modelOverseer) {
+        //register validators
+        validators.register('idFormat', idFormatValidator());
+        validators.register('testIdFormat', testidFormatValidator());
+        validators.register('testIdAvailable', idAvailableValidator(modelOverseer), true);
+    }
+
+    /**
+     * Validates the provided model
+     * @param {Object} model
+     * @throws {Error} if the model is not valid
+     */
+    function validateModel(model) {
+        const identifiers = extractIdentifiers(model, qtiTypesForUniqueIds);
+        let nonUniqueIdentifiers = 0;
+        const outcomes = _.indexBy(outcomeHelper.listOutcomes(model));
+        let messageDetails = '';
+
+        _(identifiers)
+            .countBy('identifier')
+            .forEach(function (count, id) {
+                if (count > 1) {
+                    nonUniqueIdentifiers++;
+                    messageDetails += `\n${id.originalIdentifier} : ${id.type} ${id.label}`;
+                }
+            });
+        if (nonUniqueIdentifiers.length > 1) {
+            throw new Error(__('The following identifiers are not unique accross the test : %s', messageDetails));
+        }
+
+        _.forEach(model.testParts, function (testPart) {
+            _.forEach(testPart.assessmentSections, function (assessmentSection) {
+                _.forEach(assessmentSection.rubricBlocks, function (rubricBlock) {
+                    const feedbackBlock = qtiElementHelper.lookupElement(
+                        rubricBlock,
+                        'rubricBlock.div.feedbackBlock',
+                        'content'
+                    );
+                    if (feedbackBlock && !outcomes[feedbackBlock.outcomeIdentifier]) {
+                        throw new Error(
+                            __(
+                                'The outcome "%s" does not exist, but it is referenced by a feedback block!',
+                                feedbackBlock.outcomeIdentifier
+                            )
```

Comment:
Translatable string parser can't identify multiline strings, please, if you have the possibility, make this expression in 1 line, like it was before refactoring. Thx!